MediaBrowser.Api/BaseApiService.cs

```diff
@@ -298,11 +297,24 @@ protected string GetPathValue(int index)
             var pathInfo = Parse(Request.PathInfo);
             var first = pathInfo[0];
 
+            string baseUrl = ApiEntryPoint.Instance.ConfigurationManager.Configuration.BaseUrl;
+
             // backwards compatibility
-            if (string.Equals(first, "mediabrowser", StringComparison.OrdinalIgnoreCase) ||
-                string.Equals(first, "emby", StringComparison.OrdinalIgnoreCase))
+            if (baseUrl.Length == 0
+                && (string.Equals(first, "mediabrowser", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(first, "emby", StringComparison.OrdinalIgnoreCase)))
+            {
+                index++;
+            }
+            else if (string.Equals(first, baseUrl))
             {
                 index++;
+                var second = pathInfo[1];
+                if (string.Equals(second, "mediabrowser", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(second, "emby", StringComparison.OrdinalIgnoreCase))
+                {
+                    index++;
+                }
```

Comment:
this looks to be copied twice, can we extract it to a common method, probably placed in `ApiEntryPoint.Instance.ConfigurationManager.Configuration`?

Reply:
I don't think putting this method in the configuration class is a good idea. These 2 classes share a lot of code, and IMO de-duplicating this is out of the scope of this PR.